testdata/cheats/Roll.t.sol

```diff
@@ -19,11 +19,11 @@ contract RollTest is DSTest {
     }
 
     function testRollHash() public {
-        assertEq(blockhash(block.number), keccak256(abi.encodePacked(block.number)), "initial block hash is incorrect");
+        assertEq(blockhash(block.number), 0x0, "initial block hash is incorrect");
```

Comment:
Is this right? Shouldn't bloc hash of 0 when the current block is less than 256 blocks away (this case) be non zero?

Reply:
In case you are producing a block, the current block hash is not existing.

in geth it is `< upper` and not `<= upper`: https://github.com/ethereum/go-ethereum/blob/594e32166269eed4f5cb8270bba99fa234a41606/core/vm/instructions.go#L449

Reply:
ACK - I had forgotten about this.